test/browser_tests/conf.js

```diff
@@ -203,17 +205,20 @@ var config = {
       windowHeightPx = Number( windowSizeArray[1] );
     }
 
-    browser.driver.manage().window().setSize(
-      windowWidthPx,
-      windowHeightPx
-    );
-
-    // Set default windowSize parameter equal to the value in settings.js.
-    browser.params.windowWidth = windowWidthPx;
-    browser.params.windowHeight = windowHeightPx;
-    browser.params.windowSize = String( windowWidthPx ) +
-                                ',' + String( windowHeightPx );
-
+    // Calling setSize with headless chromeDriver doesn't work properly if
+    // the requested size is larger than the available screen size.
+    if ( !envvars.HEADLESS_CHROME_BINARY ) {
+      browser.driver.manage().window().setSize(
+        windowWidthPx,
+        windowHeightPx
+      );
+
+      // Set default windowSize parameter equal to the value in settings.js.
+      browser.params.windowWidth = windowWidthPx;
+      browser.params.windowHeight = windowHeightPx;
+      browser.params.windowSize = String( windowWidthPx ) +
```

Comment:
Thanks, that's a good suggestion. I didn't modify this code at all, just modified the indent here to only get called if not running in headless mode.

Using `--window-size` is a good suggestion. In the existing setup, we always pass a window size if not provided (based on a default set in `environment.windowWidthPx` and `environment.windowHeightPx`). I'm not sure what the behavior is if you don't pass a size at all; presumably it opens at the browser default size.

I can definitely see wanting/needing to run tests against particular sizes, to test, say, mobile rendering. So there might be some more experimentation needed here to see if you can set a size as long as it's below whatever the "maximum window size" is on Travis, or if setting any size is unsupported as long as the use of Xvfb is needed.